pkg/operators/clustermanager/controllers/crdstatuccontroller/crd_status_controller.go

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2022 Contributors to the Open Cluster Management project
+ */
+
+package crdstatuccontroller
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	operatorinformer "open-cluster-management.io/api/client/operator/informers/externalversions/operator/v1"
+	operatorlister "open-cluster-management.io/api/client/operator/listers/operator/v1"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	"open-cluster-management.io/registration-operator/pkg/helpers"
+	"open-cluster-management.io/registration-operator/pkg/operators/clustermanager/controllers/migrationcontroller"
+)
+
+var (
+	// removed CRD StoredVersions
+	removedCRDStoredVersions = map[string]string{
+		"managedclustersets.cluster.open-cluster-management.io":        "v1beta1",
+		"managedclustersetbindings.cluster.open-cluster-management.io": "v1beta1",
+	}
+	// latest required CRD StoredVersions
+	requiredCRDStoredVersion = map[string]string{
+		"managedclustersets.cluster.open-cluster-management.io":        "v1beta2",
+		"managedclustersetbindings.cluster.open-cluster-management.io": "v1beta2",
+	}
+)
+
+type crdStatusController struct {
```

Comment:
1. I think clustermanager controller is used to apply basic resources but crdStatusController works only after cluster manager applied succeed. 
2. I do not want crdStatusController  affect `Applied condition` in clustermanager, and if it fails, it will just need to try reconcile its own logic, do not need to apply other resousces in clustermanager controller logic.